### PR TITLE
wdio-allure-reporter: generate typings

### LINKED ISSUE #3457: missing typings for wdio-allure-reporter
**Environment (please complete the following information):**
 - **WebdriverIO version:** 5.4.7
 - **Additional wdio packages used (if applicable):** @wdio/allure-reporter 5.4.4
- using TypeScript

**Describe the bug**
`import { addFeature } from "@wdio/allure-reporter"`
getting error
`"Could not find a declaration file for module '@wdio/allure-reporter'. '@wdio/allure-reporter/build/index.js' implicitly has an 'any' type."`

**Expected behavior**
there should be typings for allure-reporter

### PR DESCRIPTION
**Pull request by mgrybyk**
## Proposed changes

generate typings for wdio-allure-reporter

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

fix #3457

### CODE REVIEW

File: scripts/type-generation/generate-typings-utils.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+
+const changeType = (text) => {
```
Comment (mgrybyk) [inline scripts/type-generation/generate-typings-utils.js]: Just moved from webdriverio-generate-typings.js

File: scripts/type-generation/generate-typings-utils.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    }
+    default: {
+        text = text.toLowerCase()
+    }
+
+    }
+    return text
+}
+
+const getTypes = (types, alwaysType) => {
```
Comment (mgrybyk) [inline scripts/type-generation/generate-typings-utils.js]: Just moved from webdriverio-generate-typings.js

File: scripts/type-generation/generate-typings-utils.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    if (types === '' && !alwaysType) {
+        types = 'undefined'
+    } else if (types === '*' || (types === '' && alwaysType)) {
+        types = 'any'
+    }
+
+    return types
+}
+
+const buildCommand = (commandName, commandTags, indentation = 0) => {
```
Comment (mgrybyk) [inline scripts/type-generation/generate-typings-utils.js]: Moved from webdriverio-generate-typings.js, added indentation param and removing default values from param names

File: scripts/type-generation/generate-typings-utils.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        if (types && types.length === 0 && string.includes('{*}')) {
+            types.push('*')
+        }
+        if (type === 'param') {
+            let commandTypes = getTypes(types, true)
+
+            // skipping param with dot in name that stands for Object properties description, ex: attachmentObject.name
+            if (name.indexOf('.') < 0) {
+                // get rid from default values from param name, ex: [paramName='someString'] will become paramName
+                allParameters.push(`${name.split('[').pop().split('=')[0]}${optional ? '?' : ''}: ${commandTypes}`)
```
Comment (mgrybyk) [inline scripts/type-generation/generate-typings-utils.js]: get rid from default values from param name, doesn't affect existing functionality.

File: scripts/type-generation/allure-reporter-generate-typings.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+            }
+        }
+    })
+
+    return commands
+}
+
+const allureCommands = gatherCommands(path.join(__dirname + '/../../packages/wdio-allure-reporter/src/index.js'))
+
+const templatePath = path.join(__dirname, '../templates', 'allure-reporter.tpl.d.ts')
```
Comment (christian-bromann) [inline scripts/type-generation/allure-reporter-generate-typings.js]: This file is missing

Comment (mgrybyk) [reply scripts/type-generation/allure-reporter-generate-typings.js]: taking a look. Have I broken something?

Comment (mgrybyk) [reply scripts/type-generation/allure-reporter-generate-typings.js]: totally my fault, looks like I forgot to commit and push template file

Comment (mgrybyk) [reply scripts/type-generation/allure-reporter-generate-typings.js]: @christian-bromann file looks like this:
```
declare namespace AllureReporter {
    // ... AllureReporter commands ...
}

declare module "@wdio/allure-reporter" {
    export default AllureReporter
}
```

can do PR tomorrow

Comment (mgrybyk) [reply scripts/type-generation/allure-reporter-generate-typings.js]: @christian-bromann there was some magic with .gitignore

Can't image how is that webdriver.tpl.d.ts and webdriver.tpl.d.ts are not ignored but allure-reporter.tpl.d.ts is ignored

**Pull request merged**